src/nrnoc/treeset.cpp

```diff
@@ -1997,46 +1990,70 @@ static void nrn_matrix_node_alloc(void) {
         }
         /*printf(" %d extracellular nodes\n", extn);*/
         neqn += extn;
-        nt->_sp13_rhs = (double*) ecalloc(neqn + 1, sizeof(double));
-        nt->_sp13mat = spCreate(neqn, 0, &err);
-        if (err != spOKAY) {
-            hoc_execerror("Couldn't create sparse matrix", (char*) 0);
-        }
+        nt->_sparse_rhs = (double*) ecalloc(neqn + 1, sizeof(double));
+        nt->_sparseMat = new Eigen::SparseMatrix<double, Eigen::RowMajor>(neqn, neqn);
         for (in = 0, i = 1; in < nt->end; ++in, ++i) {
             nt->_v_node[in]->eqn_index_ = i;
             if (nt->_v_node[in]->extnode) {
                 i += nlayer;
             }
         }
         for (in = 0; in < nt->end; ++in) {
-            int ie, k;
+            const Node* nd = nt->_v_node[in];
+            const Extnode* nde = nd->extnode;
+            const Node* pnd = nt->_v_parent[in];
+            int i = nd->eqn_index_;
+            nt->_sparseMat->coeffRef(i - 1, i - 1) = 0.;
+            if (nde) {
+                for (int ie = 0; ie < nlayer; ++ie) {
+                    int k = i + ie + 1;
+                    nt->_sparseMat->coeffRef(k - 1, k - 1) = 0.;
+                    nt->_sparseMat->coeffRef(k - 1, k - 2) = 0;
+                    nt->_sparseMat->coeffRef(k - 2, k - 1) = 0;
+                }
+            }
+            if (pnd) {
+                int j = pnd->eqn_index_;
+                nt->_sparseMat->coeffRef(j - 1, i - 1);
+                nt->_sparseMat->coeffRef(i - 1, j - 1);
+                if (nde && pnd->extnode)
+                    for (int ie = 0; ie < nlayer; ++ie) {
+                        int kp = j + ie + 1;
+                        int k = i + ie + 1;
+                        nt->_sparseMat->coeffRef(kp - 1, k - 1) = 0;
+                        nt->_sparseMat->coeffRef(k - 1, kp - 1) = 0;
+                    }
+            }
+        }
+        // nt->_sparseMat->makeCompressed();
+        for (in = 0; in < nt->end; ++in) {
```

Comment:
This is a copy of the code below for allocating values.
`sparse13` is more or less a linked list of single value so when we get pointer afterward they are fix.
`Eigen` is a kind of vector of values. So can be reallocate. So we reserve all the values rights now and get the pointer afterward.